### ISSUE TITLE
Top-level promises catching is problematic when tests are failing

### ISSUE THREAD

**Issue by eoger** [mozilla/fxa-auth-db-mysql#321]
Top-level promises catching is problematic when tests are failing.

Let's say this assertion fails:
https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L876

This will throw and we'll end up here:
https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L895

And on my terminal, mocha will print that the following assertion is failing:
https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L41

The only way to know that the 1st assertion failed is open the chrome debugger and step in the file line by line.
I think this could be a **super-valuable** good-first-bug for an external contributor :)

**Comment by vladikoff** [comment]
@deeptibaghel , @eoger will mentor you for this bug! should be easy!

**Comment by deeptibaghel** [comment]
Sure @vladikoff , @eoger , happy to take this up :)

**Comment by eoger** [comment]
This is an example of problematic pattern:

https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L438-L451

Here you can see that if anything fails "above" that line 450 catch, we'll still end up calling `testNotFound(err)`.
What we want here is only to test `client.postThen(` on the line 443 is rejecting.
That means we should chain our promises like this:

https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L280-L288
(Notice the `.then` right after the call to `.headThen`).

TLDR: the top level "scope" should only have `.then`s, so mocha can catch these failures. (feel free to ping me on IRC if that last sentence didn't make sense)

**Comment by vladikoff** [comment]
@deeptibaghel you need to break tests on purpose to see this issue

**Comment by deeptibaghel** [comment]
> @deeptibaghel you need to break tests on purpose to see this issue

Yep got it, i experienced this while i was working on Buffer issue, it was returning lot of "Not found"s instead of actual failure message.

**Comment by deeptibaghel** [comment]
@eoger while working on the issue i found that this end point is not defined in api:
https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L802

Also used here
https://github.com/mozilla/fxa-auth-db-mysql/blob/509277957430d45390ec607406541f78d0d1b86a/db-server/test/backend/remote.js#L846

is that a bug ?

**Comment by philbooth** [comment]
@deeptibaghel, good catch!

That endpoint got removed recently, looks like we missed those test cases when removing it. If you get rid of the `+ '/verified'` bit at the end, it should work correctly! (hopefully)

**Comment by eoger** [comment]
@deeptibaghel  4808a1c47901caab28a709c58a31bd531a7c2d9c was just merged and I expect it to conflict with your future PR quite a bit, so when you're done feel free to post your patch as-is and I'll rebase and resolve the conflicts (and keep your name on it of course!).

**Comment by vladikoff** [comment]
Fixed in https://github.com/mozilla/fxa-auth-db-mysql/pull/325
